lib/msf/core/module/platform_list.rb

```diff
@@ -32,88 +29,84 @@ def self.transform(src)
   # Create an instance from an array
   #
   def self.from_a(ary)
-    self.new(*ary)
+    new(*ary)
   end
 
   def index(needle)
-    self.platforms.index(needle)
+    platforms.index(needle)
   end
 
   #
-  # Constructor, takes the entries are arguments
+  # Constructor, takes the entries as arguments
   #
   def initialize(*args)
-    self.platforms = [ ]
+    self.platforms = []
 
-    args.each { |a|
-      if a.kind_of?(String)
+    args.each do |a|
+      if a.is_a?(String)
         platforms << Msf::Module::Platform.find_platform(a)
-      elsif a.kind_of?(Range)
+      elsif a.is_a?(Range)
         b = Msf::Module::Platform.find_platform(a.begin)
         e = Msf::Module::Platform.find_platform(a.end)
 
         children = b.superclass.find_children
-        r        = (b::Rank .. e::Rank)
-        children.each { |c|
-          platforms << c if r.include?(c::Rank)
-        }
+        r = (b::Rank..e::Rank)
+        platforms << children.select { |c| r.include?(c::Rank) }
```

Comment:
iirc I don't think this is semantically equivalent but I might be wrong

```
platforms = []
children = ['a', 'b']
children.each { |c|
  platforms << c if true
}

>> platforms
=> ["a", "b"]
```

```
platforms = []
children = ['a', 'b']
platforms << children.select { |c| true }

>> platforms
=> [["a", "b"]]
```

I think you'd either want the old pattern or `.concat`

```
platforms = []
children = ['a', 'b']
platforms.concat(children.select { |c| true })

>> platforms
=> ["a", "b"]
```

That aside, I think this might be a dead code branch - I'm not sure if this method is called with ranges these days